### ISSUE TITLE
Warn when throwing away the results of known-pure functions

### ISSUE THREAD

**Issue by leeavital** [palantir/tslint#1135]
#### TypeScript code being linted

``` ts
let actions: number[];
...
actions.concat()
```
#### Actual behavior

The code passes.
#### Expected behavior

TSLint warns you that the you're throwing away the result of `actions.concat`

I'm not sure if this is out of scope for tslint, but a common mistake is to assume that `.concat()` modifies the state of `this` because `concat`  is poorly named and many other languages do the stateful thing. Same story for `Array.prototype.push`.

**Comment by leeavital** [comment]
My point about `push` was poorly communicated. I mean that you could also write a rule that prevents you from doing:

````typescript
let e = arr.push(4);
`````

**Comment by karol-majewski** [comment]
@leeavital As of now, you can use the `no-expression-statement` rule from [tslint-immutable](https://github.com/jonaskello/tslint-immutable#no-expression-statement) and `no-map-without-usage` from [tslint-clean-code](https://github.com/Glavin001/tslint-clean-code).

**Comment by JoshuaKGoldberg** [comment]
Note: per #4534, this issue will be closed in less than a month if no PR is sent to add the rule. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for new rules. See #4534. 😱

If you'd like to see this rule implemented, you have two choices:
* **Recommended:** Use [typescript-eslint](https://typescript-eslint.io) to lint your TypeScript code with ESLint ✅
* **Legacy:** Write it as a [custom TSLint rule](https://palantir.github.io/tslint/develop/custom-rules/) 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
